Source/debug.cpp

```diff
@@ -191,8 +192,11 @@ std::string DebugCmdTakeGoldCheat(const string_view parameter)
 std::string DebugCmdWarpToLevel(const string_view parameter)
 {
 	Player &myPlayer = *MyPlayer;
-	auto level = atoi(parameter.data());
-	if (level < 0 || level > (gbIsHellfire ? 24 : 16))
+	const ParseIntResult<int> parsedParam = ParseInt<int>(parameter, /*min=*/0);
+	if (!parsedParam.ok())
+		return "Specify the level number";
+	const int level = parsedParam.value;
+	if (level > (gbIsHellfire ? 24 : 16))
 		return StrCat("Level ", level, " is not known. Do you want to write a mod?");

```

Comment:
This one is getting into the weeds a bit but given status allows discriminating between failure cases you can keep the old behaviour of returning one message for out-of-range values while also putting a distinct message for missing/not a number values.
```suggestion
	const ParseIntResult<int> parsedParam = ParseInt<int>(parameter, /*min=*/0, /*max=*/(gbIsHellfire ? 24 : 16));
	if (!parsedParam.ok()) {
		switch (parsedParam.status) {
		case ParseIntStatus::ParseError:
			return "Specify the level number";
		case ParseIntStatus::OutOfRange:
			return StrCat("Level ", level, " is not known. Do you want to write a mod?");
		}
	}
	const int level = parsedParam.value;
```

There's quite a few places where the value is later checked if it's in bounds which could use a similar approach. I'd leave this for a later refactor (I'm not really happy about using a switch for this...), it'd be a nice activity if `ParseIntResult` is made more like a `std::expected` specialisation.

Reply:
This is a non-exhaustive switch, so it will result a warning.
Let's keep further improvements for a separate PR!